building-and-deploying-a-fastapi-app.md

```diff
@@ -0,0 +1,541 @@
+# Building a FastAPI Application and Deploying to Okteto
+
+
+In this tutorial, you'll learn how to develop a CRUD API with [FastAPI](https://fastapi.tiangolo.com) and deploy the application to [Okteto](https://okteto.com).
+
+You'll start by building the application, after which you will now deploy it to Okteto.
+
+## What is FastAPI
+
+FastAPI is a modern Python web framework designed for building fast and efficient backend applications. It comes with built-in support for data validation, authentication, and interactive API documentation powered by OpenAPI and Swagger.
+
+## What is Okteto?
+
+Okteto is a tool used to develop and accelerate the development workflow of the Kubernetes application.
+
+## Initial Setup
+
+Start by creating a new folder to hold your project called 'fastapi-crud':
+
+```sh
+$ mkdir fastapi-crud
+$ cd fastapi-crud
+```
+
+Next, create and activate a virtual environment in the project folder:
+
+```
+$ python3.9 -m venv venv
+$ source venv/bin/activate
+$ export PYTHONPATH=$PWD
+```
+
+Next, create the following files and folder:
+
+```
+├── app
+│       ├── __init__.py
+│       ├── model.py
+│       ├── api.py
+├──main.py
+└── requirements.txt
+```
+
+In your *requirements.txt* file, add the following dependencies:
+
+```
+fastapi==0.62.0
+uvicorn==0.13.1
+```
+
+The first dependency, `fastapi,` is the framework on which your application will be built. The second dependency, `uvicorn`, is an Asynchronous Gateway Server Interface (ASGI) that enables us to run our FastAPI application.
+
+Install the dependencies:
+
+```sh
+(venv)$ pip install -r requirements.txt
+```
+
+With the installation complete, define a base route in `app/api.py`:
+
+```py
+from fastapi import FastAPI
+
+app = FastAPI()
+
+@app.get("/", tags=["Home"])
+def get_root() -> dict:
+    returrn {
+        "message": "Welcome to the okteto's app."
+    }
+```
+
+You started by importing the FastAPI class from the `fastapi` package in the code block above. Then you created an instance of the class in the variable `app`. 
+
+Next, you defined a `GET` route on __"/"__ which is handled by the `get_root()` function.
+
+
+In the `main.py` file, define an entry point for running the application:
+
+```py
+import uvicorn
+
+if __name__ == "__main__":
+    uvicorn.run("app.api:app", host="0.0.0.0", port=8080, reload=True)
+```
+
+In the code block above, you imported the uvicorn package itself. Under the initializer block, you invoked the `fun` method, which takes the location of FastAPI's instance, the host, port, and the reload boolean value.
```

Comment:
```suggestion
In the code block above, you imported the uvicorn package itself. Under the initializer block, you invoked the `run` method, which takes the location of FastAPI's instance, the host, port, and the reload boolean value.
```

Reply:
can you explain a bit why the values were chosen?

Reply:
Fixed.